src/app/components/core/configuration-manager/configuration-manager.component.ts

```diff
@@ -34,159 +31,86 @@ export class ConfigurationManagerComponent implements OnInit {
-                  description: element.description,
-                  hasChildren: true,
-                  children: []
-                });
-              } else {
-                event.node.data.children.push({
-                  id: element.key,
-                  name: element.description,
-                  description: element.description,
-                  hasChildren: true,
-                  children: []
-                });
-              }
-              this.tree.treeModel.update();
-            });
-          }, error => {
-            if (error.status === 0) {
-              console.log('service down ', error);
-            } else {
-              this.alertService.error(error.statusText);
-            }
-          });
+  updateIdAndNameInTreeObject(tree: any) {
+    // Iterate through the array
+    tree.forEach((node: any) => {
+      // add key as Id and displayName/description as a name in the tree object
+      node.id = node.key;
+      node.name = (node.hasOwnProperty('displayName')) ? node.displayName : node.description;
+      node.hasChildren = false;
+      // If the object has 'children' property recurse 
+      if (Array.isArray(node.children) && node.children.length > 0) {
+        node.hasChildren = true;
+        this.updateIdAndNameInTreeObject(node.children);
+      }
+    })
+    return tree;
+  }
+
+  onTreeLoad(tree: TreeComponent): void {
+    if (this.selectedNode.id) {
+      this.getCategory(this.selectedNode.id, this.selectedNode.description);
+    }
+    const child = tree.treeModel.nodes[0]?.children[0];
+    if (child && !this.selectedNode.id) {
+      const firstRootChild = tree.treeModel.getNodeById(child.id);
+      firstRootChild.setActiveAndVisible();
+      this.selectedNode = {id: firstRootChild.data.id, description: firstRootChild.data.description};
+      this.getCategory(firstRootChild.data.id, firstRootChild.data.description);
     }
   }
 
-  public onNodeActive(event) {
-    this.getCategory(event.node.data.id, event.node.data.description);
-  }
-
-  public toggleDropDown() {
-    const dropDown = document.querySelector('#dropdown');
-    dropDown.classList.toggle('is-active');
-  }
-
-  public resetAllFilters() {
-    this.selectedRootCategory = 'General';
-    this.getRootCategories(true);
+  public onNodeActive(tree: TreeComponent) {
+    const rootId = tree.treeModel.focusedNodeId?.toString();
+    // In case of root node is in ['Advanced', 'General', 'Utilities'], 
+    // Expand the group and select first child
+    if (['Advanced', 'General', 'Utilities'].includes(rootId)) {
+      if (tree?.treeModel) {
+        const nodes = tree.treeModel.nodes;
+        const node = nodes.find(c => c.id == rootId)?.children[0];
+        const firstChild = tree.treeModel.getNodeById(node.id);
+        firstChild.setActiveAndVisible();
+        this.selectedNode = {id: node.id, description: node.description};
+        this.getCategory(node.id, node.description);
+      }
+    } else {
```

Comment:
why the if (tree?.treeModel) { check is not required here, if in the if block above?